openpype/hosts/max/api/preview_animation.py

```diff
@@ -0,0 +1,309 @@
+import logging
+import contextlib
+from pymxs import runtime as rt
+from .lib import get_max_version, render_resolution
+
+log = logging.getLogger("openpype.hosts.max")
+
+
+@contextlib.contextmanager
+def play_preview_when_done(has_autoplay):
+    """Function to set preview playback option during
+       context
+
+    Args:
+        has_autoplay (bool): autoplay during creating
+            preview animation
+    """
+    current_playback = rt.preferences.playPreviewWhenDone
+    try:
+        rt.preferences.playPreviewWhenDone = has_autoplay
+        yield
+    finally:
+        rt.preferences.playPreviewWhenDone = current_playback
+
+
+@contextlib.contextmanager
+def viewport_camera(camera):
+    """Function to set viewport camera during context
+    ***For 3dsMax 2024+
+    Args:
+        camera (str): viewport camera for review render
+    """
+    original = rt.viewport.getCamera()
+    has_autoplay = rt.preferences.playPreviewWhenDone
+    if not original:
+        # if there is no original camera
+        # use the current camera as original
+        original = rt.getNodeByName(camera)
+    review_camera = rt.getNodeByName(camera)
+    try:
+        rt.viewport.setCamera(review_camera)
+        rt.preferences.playPreviewWhenDone = False
+        yield
+    finally:
+        rt.viewport.setCamera(original)
+        rt.preferences.playPreviewWhenDone = has_autoplay
```

Comment:
Remove the playPreviewWhenDone logic in favor of `play_preview_when_done(has_autoplay)`
```suggestion
@contextlib.contextmanager
def viewport_camera(camera):
    """Function to set viewport camera during context
    ***For 3dsMax 2024+
    Args:
        camera (str): viewport camera for review render
    """
    original = rt.viewport.getCamera()
    if not original:
        # if there is no original camera
        # use the current camera as original
        original = rt.getNodeByName(camera)
    review_camera = rt.getNodeByName(camera)
    try:
        rt.viewport.setCamera(review_camera)
        yield
    finally:
        rt.viewport.setCamera(original)
```
Also with this removed this again looks like a fine library function.